EventGenerator/src/DIOGenerator_tool.cc

```diff
@@ -0,0 +1,68 @@
+#include "art/Utilities/ToolMacros.h"
+
+#include "CLHEP/Random/RandPoissonQ.h"
+#include "CLHEP/Random/RandGeneral.h"
+
+#include "EventGenerator/inc/ParticleGeneratorTool.hh"
+
+#include "DataProducts/inc/PDGCode.hh"
+#include "MCDataProducts/inc/GenId.hh"
+#include "Mu2eUtilities/inc/RandomUnitSphere.hh"
+#include "Mu2eUtilities/inc/BinnedSpectrum.hh"
+#include "GlobalConstantsService/inc/GlobalConstantsHandle.hh"
+#include "GlobalConstantsService/inc/ParticleDataTable.hh"
+#include "GlobalConstantsService/inc/PhysicsParams.hh"
+
+namespace mu2e {
+  class DIOGenerator : public ParticleGeneratorTool {
+  public:
+    explicit DIOGenerator(Parameters const& conf) :
+      _pdgId(PDGCode::e_minus),
+      _mass(GlobalConstantsHandle<ParticleDataTable>()->particle(_pdgId).ref().mass().value()),
+      _genId(GenId::DIOGenTool),
+      _spectrumVariable(parseSpectrumVar(conf().physics().spectrumVariable())),
+      _spectrum(BinnedSpectrum(conf().physics()))
+    {
+
+    }
+      void generate(std::unique_ptr<GenParticleCollection>& out, const IO::StoppedParticleF& stop) override;
+
+    void setEngine(art::RandomNumberGenerator::base_engine_t& eng) {
+      _randomUnitSphere = new RandomUnitSphere(eng);
+      _randSpectrum = new CLHEP::RandGeneral(eng, _spectrum.getPDF(), _spectrum.getNbins());
+    }
+
+  private:
+    PDGCode::type _pdgId;
+    double _mass;
+    GenId _genId;
+
+    SpectrumVar       _spectrumVariable;
```

Comment:
Suggest to remove _spectrumVariable.  This is a DIO generator, isn't it always going to be energy? (Unless someone make a mistake in  fhicl config.)

Reply:
Done in commit 1c746446b